optimizebuildingplacement.py

```diff
@@ -0,0 +1,76 @@
+# Time Complexity : Not sure
```

Comment:
Time Complexity - 2^hw 
for BFS - O(h*w), 
Backtrack - 2^hw 
h - height and w - width of the grid